### DISCUSSION TITLE
OpenSearch Store

### DISCUSSION THREAD

**Discussion by domkun** [eclipse-rdf4j/rdf4j#4830] [General]
Hi there,

I am thinking about porting the Elasticsearch Store to [OpenSearch](https://opensearch.org). Is there any interest in having a OpenSearch Store?

I thought about copying the existing elasticsearch-store and migrate all Elasticsearch specific code to OpenSearch.

What is the current state of the Elasticsearch-Store? I think it is built upon an older version of Elasticsearch. Is it still considered to be an active component of the RDF4J project?

**Comment by hmottestad** [comment]
I feel that OpenSearch undermines the spirit of open source because it's the fork of Elasticsearch that Amazon created instead of coming to a licensing agreement with Elastic. 

As for the ElasticsearchStore, it's not going anywhere. I even started some work on porting it to the latests elasticsearch client: https://github.com/eclipse-rdf4j/rdf4j/pull/4777

**Reply by domkun** [reply]
That is a valid point. Thought it is maybe a good starting point to dig a bit deeper into RDF4J development as it is a self contained component. I think I will find another way then.

Thank you for your opinion.

**Reply by hmottestad** [reply]
Any chance you would want to help with upgrading to the new elasticsearch client?

**Reply by hmottestad** [reply]
I don't have much time to work on it at the moment, and I'm also not super familiar with the new api.

**Reply by domkun** [reply]
I am also not too familiar with the new API. I am on vacation till monday, then I will have a look at it.
